lib/active_model_serializers/adapter/json_api/deserialization.rb

```diff
@@ -140,12 +140,18 @@ def validate_payload(payload)
         # rubocop:enable Metrics/CyclomaticComplexity
 
         # @api private
-        def filter_fields(fields, options)
+        def filter_fields(hash, options)
+          hash = transform_keys(hash, options)
```

Comment:
So, `parse` should be like this?

```ruby
        # Same as parse!, but returns an empty hash instead of raising InvalidDocument
        # on invalid payloads.
        def parse(document, options = {})
          document = document.dup.permit!.to_h if document.is_a?(ActionController::Parameters)

          validate_payload(document) do |invalid_document, reason|
            yield invalid_document, reason if block_given?
            return {}
          end

          primary_data = document['data']
          attributes = primary_data['attributes'] || {}
          attributes['id'] = primary_data['id'] if primary_data['id']
          relationships = primary_data['relationships'] || {}

          attributes = transform_keys(attributes, options)
          relationships = transform_keys(relationships, options)

          attributes = filter_fields(attributes, options)
          relationships = filter_fields(relationships, options)

          hash = {}
          hash.merge!(parse_attributes(attributes, options))
          hash.merge!(parse_relationships(relationships, options))

          hash
        end
```

after removing transform_keys from `filter_fields`

it does make sense from a single-responsibility point of view -- just clarifying